### ISSUE TITLE
Python随身听-2020-10-29技术精选

### ISSUE THREAD

**Issue by de8ug** [pythonradio/pythonradio.github.io#39]
# Python随身听-2020-10-29-技术精选

> 致读者：亲爱的「Python随身听」的观众们，这是由DE8UG的人工非智能给你带来的新的一期技术精选。  
> 主要为编程初学者，开发工程师，算法工程师，数据分析师，运维，测试，运营，产品等各个岗位的Python爱好者带来Python世界的流行趋势，前沿技术。  
> 你可以挑选自己喜欢的项目尽情玩耍，任何想法欢迎留言讨论。  
> 本文的结构和内容会经常更新，每天10：24分左右发布，感谢订阅🆙和收藏☆。  
>（点击原文或到pythonradio.online网站查看可点击的文档链接）  

![](https://img2020.cnblogs.com/blog/152871/202010/152871-20201005104055252-1408177983.png)

## 🤩Python随身听-技术精选： /lrvick/youtube-dl
👉RIAA: Please go die in a fire.  

😎TOPICS: ``  

⭐️STARS：318, 今日上升数↑:196  

👉README：

youtube-dl - download videos from youtube.com or other video platforms

- [INSTALLATION](#installation)
- [DESCRIPTION](#description)
- [OPTIONS](#options)
- [CONFIGURATION](#configuration)
- [OUTPUT TEMPLATE](#output-template)
- [FORMAT SELECTION](#format-selection)
- [VIDEO SELECTION](#video-selection)
- [FAQ](#faq)
- [DEVELOPER INSTRUCTIONS](#developer-instructions)
- [EMBEDDING YOUTUBE-DL](#embedding-youtube-dl)
- [BUGS](#bugs)
- [COPYRIGHT](#copyright)

## INSTALLATION

To install it right away for all UNIX users (Linux, macOS, etc.), type:

    sudo curl -L https://yt-dl.org/downloads/latest/youtube-dl -o /usr/local/bin/youtube-dl
    sudo chmod a+rx /usr/local/bin/youtube-dl

If you do not have curl, you can alternatively use a recent wget:

    sudo wget https://yt-dl.org/downloads/latest/youtube-dl -O /usr/local/bin/youtube-dl
    sudo chmod a+rx /usr/local/bin/youtube-dl

Windows users can [download an .exe file](https://yt-dl.org/latest/youtube-dl.exe) and place it in any location on their [PATH](...

地址：[https://github.com/lrvick/youtube-dl](https://github.com/lrvick/youtube-dl)

---

## 🤩Python随身听-技术精选： /sherlock-project/sherlock
👉🔎 Hunt down social media accounts by username across social networks  

😎TOPICS: `osint,reconnaissance,linux,macos,cli,sherlock,python3,windows,redteam,tools,information-gathering`  

⭐️STARS：16435, 今日上升数↑:278  

👉README：

<p align=center>

  <img src="https://user-images.githubusercontent.com/27065646/53551960-ae4dff80-3b3a-11e9-9075-cef786c69364.png"/>

  <br>
  <span>Hunt down social media accounts by username across <a href="https://github.com/sherlock-project/sherlock/blob/master/sites.md">social networks</a></span>
  <br>
  <a target="_blank" href="https://www.python.org/downloads/" title="Python version"><img src="https://img.shields.io/badge/python-%3E=_3.6-green.svg"></a>
  <a target="_blank" href="LICENSE" title="License: MIT"><img src="https://img.shields.io/badge/License-MIT-blue.svg"></a>
  <a target="_blank" href="https://github.com/sherlock-project/sherlock/actions" title="Test Status"><img src="https://github.com/sherlock-project/sherlock/workflows/Tests/badge.svg?branch=master"></a>
  <a target="_blank" href="https://github.com/sherlock-project/sherlock/actions" title="Nightly Tests"><img src="https://github.com/sherlock-project/sherlock/workflows/Nightly/badge.svg?branch=master"></a>
  <a target="_blank" href=...

地址：[https://github.com/sherlock-project/sherlock](https://github.com/sherlock-project/sherlock)

---

## 🤩Python随身听-技术精选： /blackjack4494/yt-dlc
👉media downloader for various sites.  

😎TOPICS: `downloader,media,library,streaming`  

⭐️STARS：823, 今日上升数↑:167  

👉README：

youtube-dlc - download videos from youtube.com or other video platforms.

youtube-dlc is a fork of youtube-dl with the intention of getting features tested by the community merged in the tool faster, since youtube-dl's development seems to be slowing down. (https://web.archive.org/web/20201014194602/https://github.com/ytdl-org/youtube-dl/issues/26462)

- [INSTALLATION](#installation)
- [DESCRIPTION](#description)
- [OPTIONS](#options)
  - [Network Options:](#network-options)
  - [Geo Restriction:](#geo-restriction)
  - [Video Selection:](#video-selection)
  - [Download Options:](#download-options)
  - [Filesystem Options:](#filesystem-options)
  - [Thumbnail images:](#thumbnail-images)
  - [Verbosity / Simulation Options:](#verbosity--simulation-options)
  - [Workarounds:](#workarounds)
  - [Video Format Options:](#video-format-options)
  - [Subtitle Options:](#subtitle-options)
  - [Authentication Options:](#authentication-options)
  - [Adobe Pass Options:](#adobe-pass-options)
  - [Post-processing Options...

地址：[https://github.com/blackjack4494/yt-dlc](https://github.com/blackjack4494/yt-dlc)

---

## 🤩Python随身听-技术精选： /l1ving/youtube-dl
👉A copyright-respecting fork of youtube-dl  

😎TOPICS: ``  

⭐️STARS：1109, 今日上升数↑:184  

👉README：

youtube-dl - download videos from youtube.com or other video platforms

- [CHANGES](#changes)
- [INSTALLATION](#installation)
- [DESCRIPTION](#description)
- [OPTIONS](#options)
- [CONFIGURATION](#configuration)
- [OUTPUT TEMPLATE](#output-template)
- [FORMAT SELECTION](#format-selection)
- [VIDEO SELECTION](#video-selection)
- [FAQ](#faq)
- [DEVELOPER INSTRUCTIONS](#developer-instructions)
- [EMBEDDING YOUTUBE-DL](#embedding-youtube-dl)
- [BUGS](#bugs)
- [COPYRIGHT](#copyright)

## CHANGES 

You can view the changes made to ytdl-org/youtube-dl [here](https://github.com/l1ving/youtube-dl/compare/416da574ec0df3388f652e44f7fe71b1e3a4701f...master)

You can view the archived tags here: [youtube-dl/releases](https://github.com/l1ving/youtube-dl/releases)

You can view the archived unmerged pull requests here: [youtube-dl/tree/archive/recovered-github-prs](https://github.com/l1ving/youtube-dl/tree/archive/recovered-github-prs)

## INSTALLATION

To install it right away for all UNIX users (Linux, macOS, etc.), typ...

地址：[https://github.com/l1ving/youtube-dl](https://github.com/l1ving/youtube-dl)

---

## 🤩Python随身听-技术精选： /EssayKillerBrain/EssayKiller_V2
👉基于开源GPT2.0的初代创作型人工智能 | 可扩展、可进化  

😎TOPICS: ``  

⭐️STARS：1079, 今日上升数↑:118  

👉README：

## EssayKiller

通用型议论文创作人工智能框架，仅限交流与科普。

Bilibili视频地址：https://www.bilibili.com/video/BV1pr4y1w7uM/

### 项目简介
EssayKiller是基于OCR、NLP领域的最新模型所构建的生成式文本创作AI框架，目前第一版finetune模型针对高考作文（主要是议论文），可以有效生成符合人类认知的文章，多数文章经过测试可以达到正常高中生及格作文水平。

| 项目作者        | 主页1           | 主页2  | 主页3 |
| ------------- |:-------------:|:----:|:---:|
| 图灵的猫       | [知乎](https://www.zhihu.com/people/dong-xi-97-29) |[B站](https://space.bilibili.com/371846699) | [Youtube](https://www.youtube.com/channel/UCoEVP6iTw5sfozUGLLWJyDg/featured) |

**致谢**

感谢开源作者[@imcaspar](https://github.com/imcaspar)  提供GPT-2中文预训练框架与数据支持。
感谢[@白小鱼博士](https://www.zhihu.com/people/youngfish42) 、[@YJango博士](https://www.zhihu.com/people/YJango) 、[@画渣花小烙](https://space.bilibili.com/402576555)、[@万物拣史](https://space.bilibili.com/328531988/) 、[@柴知道](https://space.bilibili.com/26798384/)、[@风羽酱-sdk](https://space.bilibili.com/17466521/)、[@WhatOnEarth](https://space.bilibili.com/410527811/)、[@这知识好冷](https://space.bilibili.com/403943112/)、[@科技狐](https://space.bilibili.com/404334...

地址：[https://github.com/EssayKillerBrain/EssayKiller_V2](https://github.com/EssayKillerBrain/EssayKiller_V2)

---

## 🤩Python随身听-技术精选： /scastillo/not-youtube-dl
👉This is not youtube-dl  

😎TOPICS: ``  

⭐️STARS：704, 今日上升数↑:152  

👉README：

this is not youtube-dl - it does not download videos from youtube.com or other video platforms

- [INSTALLATION](#installation)
- [DESCRIPTION](#description)
- [OPTIONS](#options)
- [CONFIGURATION](#configuration)
- [OUTPUT TEMPLATE](#output-template)
- [FORMAT SELECTION](#format-selection)
- [VIDEO SELECTION](#video-selection)
- [FAQ](#faq)
- [DEVELOPER INSTRUCTIONS](#developer-instructions)
- [EMBEDDING YOUTUBE-DL](#embedding-youtube-dl)
- [BUGS](#bugs)
- [COPYRIGHT](#copyright)

## INSTALLATION

To install it right away for all UNIX users (Linux, macOS, etc.), type:

    sudo curl -L https://yt-dl.org/downloads/latest/youtube-dl -o /usr/local/bin/youtube-dl
    sudo chmod a+rx /usr/local/bin/youtube-dl

If you do not have curl, you can alternatively use a recent wget:

    sudo wget https://yt-dl.org/downloads/latest/youtube-dl -O /usr/local/bin/youtube-dl
    sudo chmod a+rx /usr/local/bin/youtube-dl

Windows users can [download an .exe file](https://yt-dl.org/latest/youtube-dl.exe) and place it in any l...

地址：[https://github.com/scastillo/not-youtube-dl](https://github.com/scastillo/not-youtube-dl)

---

## 🤩Python随身听-技术精选： /vinta/awesome-python
👉A curated list of awesome Python frameworks, libraries, software and resources  

😎TOPICS: `awesome,python,collections,python-library,python-framework,python-resources`  

⭐️STARS：88175, 今日上升数↑:101  

👉README：

A curated list of awesome Python frameworks, libraries, software and resources.

Inspired by [awesome-php](https://github.com/ziadoz/awesome-php).

- [Awesome Python](#awesome-python)
    - [Admin Panels](#admin-panels)
    - [Algorithms and Design Patterns](#algorithms-and-design-patterns)
    - [ASGI Servers](#asgi-servers)
    - [Asynchronous Programming](#asynchronous-programming)
    - [Audio](#audio)
    - [Authentication](#authentication)
    - [Build Tools](#build-tools)
    - [Built-in Classes Enhancement](#built-in-classes-enhancement)
    - [Caching](#caching)
    - [ChatOps Tools](#chatops-tools)
    - [CMS](#cms)
    - [Code Analysis](#code-analysis)
    - [Command-line Interface Development](#command-line-interface-development)
    - [Command-line Tools](#command-line-tools)
    - [Compatibility](#compatibility)
    - [Computer Vision](#computer-vision)
    - [Concurrency and Parallelism](#concurrency-and-parallelism)
    - [Configuration](#configuration)
    - [Cryptography](#cryptography)
   ...

地址：[https://github.com/vinta/awesome-python](https://github.com/vinta/awesome-python)

---

## 🤩Python随身听-技术精选： /donnemartin/system-design-primer
👉Learn how to design large-scale systems. Prep for the system design interview.  Includes Anki flashcards.  

😎TOPICS: `programming,development,design,design-system,system,design-patterns,web,web-application,webapp,python,interview,interview-questions,interview-practice`  

⭐️STARS：110264, 今日上升数↑:103  

👉README：

*[English](README.md) ∙ [日本語](README-ja.md) ∙ [简体中文](README-zh-Hans.md) ∙ [繁體中文](README-zh-TW.md) | [العَرَبِيَّة‎](https://github.com/donnemartin/system-design-primer/issues/170) ∙ [বাংলা](https://github.com/donnemartin/system-design-primer/issues/220) ∙ [Português do Brasil](https://github.com/donnemartin/system-design-primer/issues/40) ∙ [Deutsch](https://github.com/donnemartin/system-design-primer/issues/186) ∙ [ελληνικά](https://github.com/donnemartin/system-design-primer/issues/130) ∙ [עברית](https://github.com/donnemartin/system-design-primer/issues/272) ∙ [Italiano](https://github.com/donnemartin/system-design-primer/issues/104) ∙ [한국어](https://github.com/donnemartin/system-design-primer/issues/102) ∙ [فارسی](https://github.com/donnemartin/system-design-primer/issues/110) ∙ [Polski](https://github.com/donnemartin/system-design-primer/issues/68) ∙ [русский язык](https://github.com/donnemartin/system-design-primer/issues/87) ∙ [Español](https://github.com/donnemartin/system-design-primer/issues/136) ∙ [...

地址：[https://github.com/donnemartin/system-design-primer](https://github.com/donnemartin/system-design-primer)

---

## 🤩Python随身听-技术精选： /youtube-dl2/youtube-dl
👉Repository with the code of youtube-dl  

😎TOPICS: `youtube-dl,dmca-takedown`  

⭐️STARS：377, 今日上升数↑:76  

👉README：

youtube-dl - download videos from youtube.com or other video platforms

- [INSTALLATION](#installation)
- [DESCRIPTION](#description)
- [OPTIONS](#options)
- [CONFIGURATION](#configuration)
- [OUTPUT TEMPLATE](#output-template)
- [FORMAT SELECTION](#format-selection)
- [VIDEO SELECTION](#video-selection)
- [FAQ](#faq)
- [DEVELOPER INSTRUCTIONS](#developer-instructions)
- [EMBEDDING YOUTUBE-DL](#embedding-youtube-dl)
- [BUGS](#bugs)
- [COPYRIGHT](#copyright)

## INSTALLATION

To install it right away for all UNIX users (Linux, macOS, etc.), type:

    sudo curl -L https://yt-dl.org/downloads/latest/youtube-dl -o /usr/local/bin/youtube-dl
    sudo chmod a+rx /usr/local/bin/youtube-dl

If you do not have curl, you can alternatively use a recent wget:

    sudo wget https://yt-dl.org/downloads/latest/youtube-dl -O /usr/local/bin/youtube-dl
    sudo chmod a+rx /usr/local/bin/youtube-dl

Windows users can [download an .exe file](https://yt-dl.org/latest/youtube-dl.exe) and place it in any location on their [PATH](...

地址：[https://github.com/youtube-dl2/youtube-dl](https://github.com/youtube-dl2/youtube-dl)

---

## 🤩Python随身听-技术精选： /corkami/mitra
👉A generator of binary polyglots  

😎TOPICS: ``  

⭐️STARS：331, 今日上升数↑:53  

👉README：

## Mitra

A tool to generate binary polyglots
(files that are valid with several file formats).

Loosely named after [Μιθραδάτης](https://en.wikipedia.org/wiki/Mithridates_VI_of_Pontus),
a famous polyglot.
Pronounced `mɪtrə`.

[What's new](NEWS.md).

### How to use

`mitra.py file1.png file2.dcm` gives you a working PNG/DICOM polyglot.

Check Corkami [mini](https://github.com/corkami/pocs/tree/master/mini)
or [tiny](https://github.com/corkami/pocs/tree/master/tiny) PoCs for input files.
and the formats [repository](https://github.com/corkami/formats/tree/WIP) for some extra technical info.

## Features

It tries different layouts:
Stacks (appended data), Cavities (blank space), Parasites (comments), Zippers (mutual comments).

It returns the offsets where the payloads 'switch sizes' for multi-ciphertexts.

Ex: `Z(80-162-286)-DICOM^TIFF.be3b767b.dcm.tif` is a DICOM/TIFF zipper
where the payloads switch side at offsets `0x80`, `0x162` and `0x286`.

The `-s` option extracts the 2 payloads separately, mixed with...

地址：[https://github.com/corkami/mitra](https://github.com/corkami/mitra)

---

## 🤩Python随身听-技术精选： /alexandreborges/malwoverview
👉Malwoverview is a first response tool used for downloading and screening malware samples, suspicious URLs, IP address, domains. Malwoverview offers threat hunting information from Virus Total, Hybrid Analysis, URLHaus, Polyswarm, Malshare, Alien Vault, Malpedia, ThreatCrowd, Valhalla and it is able to scan Android devices against VT and HA.  

😎TOPICS: `malware,virustotal,hybridanalysis,polyswarm,malpedia,urlhaus,alienvault,malshare,threatcrowd,valhalla,threathunting`  

⭐️STARS：1107, 今日上升数↑:14  

👉README：

## Malwoverview

[<img alt="GitHub release (latest by date)" src="https://img.shields.io/github/v/release/alexandreborges/malwoverview?color=Red&style=for-the-badge">](https://github.com/alexandreborges/malwoverview/releases/tag/4.2) [<img alt="GitHub last commit" src="https://img.shields.io/github/last-commit/alexandreborges/malwoverview?color=Yellow&style=for-the-badge">](https://github.com/alexandreborges/malwoverview/releases) [<img alt="GitHub Release Date" src="https://img.shields.io/github/release-date/alexandreborges/malwoverview?label=Release%20Date&style=for-the-badge">](https://github.com/alexandreborges/malwoverview/releases) [<img alt="GitHub" src="https://img.shields.io/github/license/alexandreborges/malwoverview?style=for-the-badge">](https://github.com/alexandreborges/malwoverview/blob/master/LICENSE) 
[<img alt="GitHub stars" src="https://img.shields.io/github/stars/alexandreborges/malwoverview?logoColor=Red&style=for-the-badge">](https://github.com/alexandreborges/malwoverview/stargazers) [<...

地址：[https://github.com/alexandreborges/malwoverview](https://github.com/alexandreborges/malwoverview)

---

## 🤩Python随身听-技术精选： /man-group/dtale
👉Visualizer for pandas data structures  

😎TOPICS: `python27,python3,react,flask,pandas,ipython,jupyter-notebook,react-virtualized,data-analysis,data-visualization,visualization,plotly-dash,data-science,xarray`  

⭐️STARS：1567, 今日上升数↑:39  

👉README：

* [Live Demo](http://alphatechadmin.pythonanywhere.com)
* [Animated US COVID-19 Deaths By State](http://alphatechadmin.pythonanywhere.com/charts/3?chart_type=maps&query=date+%3E+%2720200301%27&agg=raw&map_type=choropleth&loc_mode=USA-states&loc=state_code&map_val=deaths&colorscale=Reds&cpg=false&animate_by=date)
* [3D Scatter Chart](http://alphatechadmin.pythonanywhere.com/charts/4?chart_type=3d_scatter&query=&x=date&z=Col0&agg=raw&cpg=false&y=%5B%22security_id%22%5D)
* [Surface Chart](http://alphatechadmin.pythonanywhere.com/charts/4?chart_type=surface&query=&x=date&z=Col0&agg=raw&cpg=false&y=%5B%22security_id%22%5D)

### What is it?

D-Tale is the combination of a Flask back-end and a React front-end to bring you an easy way to view & analyze Pandas data structures.  It integrates seamlessly with ipython notebooks & python/ipython terminals.  Currently this tool supports such Pandas objects as DataFrame, Series, MultiIndex, DatetimeIndex & RangeIndex.

### Origins

D-Tale was the product of a SAS to Pyth...

地址：[https://github.com/man-group/dtale](https://github.com/man-group/dtale)

---

## 🤩Python随身听-技术精选： /alexa/bort
👉Repository for the paper "Optimal Subarchitecture Extraction for BERT"  

😎TOPICS: ``  

⭐️STARS：159, 今日上升数↑:50  

👉README：

###### Companion code for the paper "Optimal Subarchitecture Extraction for BERT."

Bort is an optimal subset of architectural parameters for the BERT architecture, extracted by applying a fully polynomial-time approximation scheme (FPTAS) for neural architecture search. Bort has an effective (that is, not counting the embedding layer) size of 5.5\% the original BERT-large architecture, and 16\% of the net size. It is also able to be pretrained in 288 GPU hours, which is 1.2\% of the time required to pretrain the highest-performing BERT parametric architectural variant, RoBERTa-large.
It is also 7.9x faster on a CPU, and performs better than other compressed variants of the architecture, and some of the non-compressed variants; it obtains an average performance improvement of between 0.3\% and 31\%, absolute with respect to BERT-large on multiple public natural language understanding (NLU) benchmarks.

Here are the corresponding GLUE scores on the test set:

|Model|Score|CoLA|SST-2|MRPC|STS-B|QQP|MNLI...

地址：[https://github.com/alexa/bort](https://github.com/alexa/bort)

---

## 🤩Python随身听-技术精选： /twintproject/twint
👉An advanced Twitter scraping & OSINT tool written in Python that doesn't use Twitter's API, allowing you to scrape a user's followers, following, Tweets and more while evading most API limitations.  

😎TOPICS: `osint,twitter,python,scrape,tweets,elasticsearch,kibana,scrape-followers,scrape-likes,scrape-following,twint,tweep`  

⭐️STARS：8397, 今日上升数↑:14  

👉README：

## TWINT - Twitter Intelligence Tool

>No authentication. No API. No limits.

Twint is an advanced Twitter scraping tool written in Python that allows for scraping Tweets from Twitter profiles **without** using Twitter's API.

Twint utilizes Twitter's search operators to let you scrape Tweets from specific users, scrape Tweets relating to certain topics, hashtags & trends, or sort out *sensitive* information from Tweets like e-mail and phone numbers. I find this very useful, and you can get really creative with it too.

Twint also makes special queries to Twitter allowing you to also scrape a Twitter user's followers, Tweets a user has liked, and who they follow **without** any authentication, API, Selenium, or browser emulation.

### tl;dr Benefits
Some of the benefits of using Twint vs Twitter API:
- Can fetch almost __all__ Tweets (Twitter API limits to last 3200 Tweets only);
- Fast initial setup;
- Can be used anonymously and without Twitter sign up;
- **No rate limitations**.

### Limits imposed by Twi...

地址：[https://github.com/twintproject/twint](https://github.com/twintproject/twint)

---

## 🤩Python随身听-技术精选： /google-research/google-research
👉Google Research  

😎TOPICS: `machine-learning,ai,research`  

⭐️STARS：13541, 今日上升数↑:204  

👉README：

## Google Research

This repository contains code released by
[Google Research](https://research.google).

All datasets in this repository are released under the CC BY 4.0 International
license, which can be found here:
https://creativecommons.org/licenses/by/4.0/legalcode.  All source files in this
repository are released under the Apache 2.0 license, the text of which can be
found in the LICENSE file.

Because the re...

地址：[https://github.com/google-research/google-research](https://github.com/google-research/google-research)

---

## 🤩Python随身听-技术精选： /zergtant/pytorch-handbook
👉pytorch handbook是一本开源的书籍，目标是帮助那些希望和使用PyTorch进行深度学习开发和研究的朋友快速入门，其中包含的Pytorch教程全部通过测试保证可以成功运行  

😎TOPICS: `pytorch,pytorch-tutorials,pytorch-handbook,deep-learning,neural-network,machine-learning`  

⭐️STARS：12836, 今日上升数↑:13  

👉README：

### 书籍介绍
这是一本开源的书籍，目标是帮助那些希望和使用PyTorch进行深度学习开发和研究的朋友快速入门。

由于本人水平有限，在写此教程的时候参考了一些网上的资料，在这里对他们表示敬意，我会在每个引用中附上原文地址，方便大家参考。

深度学习的技术在飞速的发展，同时PyTorch也在不断更新，且本人会逐步完善相关内容。

### 版本说明
由于PyTorch版本更迭，教程的版本会与PyTorch版本，保持一致。
当前版本 1.6

### QQ 5群 

群号：894059877

扫描二维码

[点击链接加入群聊 『PyTorch Handbook 交流5群』](//jq.qq.com/?_wv=1027&k=HI1gtcVa)

1群(985896536)已满，2群(681980831) 3群(773681699)已满  4群(884017356)已满 

不要再加了

### 新福利

公众账号每日分享干货文章

### 说明

- 修改错别字请直接提issue或PR

- PR时请注意版本

- 有问题也请直接提issue

感谢

#### 第一章：PyTorch 入门

1. [PyTorch 简介](chapter1/1.1-pytorch-introduction.md)
2. [PyTorch 环境搭建](chapter1/1.2-pytorch-installation.md)
3. [PyTorch 深度学习：60分钟快速入门（官方）](chapter1/1.3-deep-learning-with-pytorch-60-minute-blitz.md)
    - [张量](chapter1/1_tensor_tutorial.ipynb)
    - [Autograd：自动求导](chapter1/2_autograd_tutorial.ipynb) 
    - [神经网络](chapter1/3_neural_networks_tutorial.ipynb)
    - [训练一个分类器](chapter1/4_cifar10_tutorial.ipynb)
    - [选读：数据并行处理（多GPU）](chapter1/5_data_parallel_tutorial.ipy...

地址：[https://github.com/zergtant/pytorch-handbook](https://github.com/zergtant/pytorch-handbook)

---

## 🤩Python随身听-技术精选： /slundberg/shap
👉A game theoretic approach to explain the output of any machine learning model.  

😎TOPICS: `interpretability,machine-learning,deep-learning,gradient-boosting,shap,shapley,explainability`  

⭐️STARS：10594, 今日上升数↑:12  

👉README：

<p align="center">
  <img src="https://raw.githubusercontent.com/slundberg/shap/master/docs/artwork/shap_header.png" width="800" />
</p>

<a href="https://travis-ci.org/slundberg/shap"><img src="https://travis-ci.org/slundberg/shap.svg?branch=master"></a>

**SHAP (SHapley Additive exPlanations)** is a game theoretic approach to explain the output of any machine learning model. It connects optimal credit allocation with local explanations using the classic Shapley values from game theory and their related extensions (see [papers](#citations) for details and citations).

### Install

Shap can be installed from either ...

地址：[https://github.com/slundberg/shap](https://github.com/slundberg/shap)

---

## 🤩Python随身听-技术精选： /ageron/handson-ml2
👉A series of Jupyter notebooks that walk you through the fundamentals of Machine Learning and Deep Learning in Python using Scikit-Learn, Keras and TensorFlow 2.  

😎TOPICS: ``  

⭐️STARS：11050, 今日上升数↑:23  

👉README：

Machine Learning Notebooks

This project aims at teaching you the fundamentals of Machine Learning in
python. It contains the example code and solutions to the exercises in the second edition of my O'Reilly book [Hands-on Machine Learning with Scikit-Learn, Keras and TensorFlow](https://www.oreilly.com/library/view/hands-on-machine-learning/9781492032632/):

<img src="https://images-na.ssl-images-amazon.com/images/I/51aqYc1QyrL._SX379_BO1,204,203,200_.jpg" title="book" width="150" />

**Note**: If you are looking for the first edition notebooks, check out [ageron/handson-ml](https://github.com/ageron/handson-ml).

#### Want to play with these notebooks online without having to install anything?
Use any of the following services.

**WARNING**: Please be aware that these services provide temporary environments: anything you do will be deleted after a while, so make sure you download any data you care about.

* **Recommended**: open this repository in [Colaboratory](https://colab.research.google...

地址：[https://github.com/ageron/handson-ml2](https://github.com/ageron/handson-ml2)

---

## 🤩Python随身听-技术精选： /AtsushiSakai/PythonRobotics
👉Python sample codes for robotics algorithms.  

😎TOPICS: `python,robotics,algorithm,path-planning,control,animation,localization,slam,cvxpy,ekf,autonomous-vehicles,autonomous-driving,mapping,autonomous-navigation,robot`  

⭐️STARS：10525, 今日上升数↑:11  

👉README：

<img src="https://github.com/AtsushiSakai/PythonRobotics/raw/master/icon.png?raw=true" align="right" width="300" alt="header pic"/>

## PythonRobotics

Python codes for robotics algorithm.

## Table of Contents
   * [What is this?](#what-is-this)
   * [Requirements](#requirements)
   * [Documentation](#documentation)
   * [How to use](#how-to-use)
   * [Localization](#localization)
      * [Extended Kalman Filter localization](#extended-kalman-filter-localization)
      * [Particle filter localization](#particle-filter-localization)
      * [Histogram filter localization](#histogram-filter-localization)
   * [Mapping](#mapping)
      * [Gaussian grid map](#gaussian-grid-map)
      * [Ray casting grid map](#ray-casting-grid-map)
      * [Lidar to grid map](#lidar-to-grid-map)
      * [k-means object clustering](#k-means-object-clustering)
      * [Rectangle fitting](#rectangle-fitting)
   * [SLAM](#slam)
      * [Iterative Closest Point (ICP) Matching](#iterative-closest-point-icp-matching)
      * [FastSLAM ...

地址：[https://github.com/AtsushiSakai/PythonRobotics](https://github.com/AtsushiSakai/PythonRobotics)

---

## 🤩Python随身听-技术精选： /pandas-profiling/pandas-profiling
👉Create HTML profiling reports from pandas DataFrame objects  

😎TOPICS: `pandas-profiling,pandas-dataframe,statistics,jupyter-notebook,exploration,data-science,python,pandas,machine-learning,artificial-intelligence,deep-learning,exploratory-data-analysis,eda,data-quality,html-report,data-exploration,data-analysis,jupyter,big-data-analytics,data-profiling`  

⭐️STARS：6166, 今日上升数↑:12  

👉README：

## Pandas Profiling

<p align="center">
  <a href="https://pandas-profiling.github.io/pandas-profiling/docs/master/rtd/">Documentation</a>
  |
  <a href="https://join.slack.com/t/pandas-profiling/shared_invite/zt-hfy3iwp2-qEJSItye5QBZf8YGFMaMnQ">Slack</a>
  | 
  <a href="https://stackoverflow.com/questions/tagged/pandas-profiling">Stack Overflow</a>
</p>

Generates profile reports from a pandas `DataFrame`. 
The pandas `df.describe()` function is great but a little basic for serious exploratory data analysis. 
`pandas_profiling` extends the pandas DataFrame with `df.profile_report()` for quick data analysis.

For each column the following statistics - if relevant for the column type - are presented in an interactive HTML report:

* **Type inference**: detect the [types](#types) of columns in a dataframe.
* **Essentials**: type, unique values, missing values
* **Quantile statistics** like minimum value, Q1, median, Q3, maximum, range, interquartile range
* **Descriptive statistics** like mean, mode, standar...

地址：[https://github.com/pandas-profiling/pandas-profiling](https://github.com/pandas-profiling/pandas-profiling)

---

## 🤩Python随身听-技术精选： /NLP-LOVE/ML-NLP
👉此项目是机器学习(Machine Learning)、深度学习(Deep Learning)、NLP面试中常考到的知识点和代码实现，也是作为一个算法工程师必会的理论基础知识。  

😎TOPICS: `nlp,machine-learning,deep-learning`  

⭐️STARS：6777, 今日上升数↑:17  

👉README：

### 项目介绍

- 此项目是**机器学习、NLP面试**中常考到的**知识点和代码实现**，也是作为一个算法工程师必会的理论基础知识。
- 既然是以面试为主要目的，亦不可以篇概全，请谅解，有问题可提出。
- 此项目以各个模块为切入点，让大家有一个清晰的知识体系。
- 此项目亦可拿来常读、常记以及面试时复习之用。
- 每一章里的问题都是面试时有可能问到的知识点，如有遗漏可联系我进行补充，结尾处都有算法的**实战代码案例**。
- 思维导图，**请关注 AIArea 公众号并回复：NLP思维导图** ，即能下载高清大图。

### 目录

- **项目持续更新中......**

| 模块     | 章节                                                         | 负责人(GitHub)                          | 联系QQ    |
| -------- | ------------------------------------------------------------ | --------------------------------------- | --------- |
| 机器学习 | [1. 线性回归(Liner Regression)](https://github.com/NLP-LOVE/ML-NLP/blob/master/Machine%20Learning/Liner%20Regression/1.Liner%20Regression.md) | [@mantchs](https://github.com/NLP-LOVE) | 448966528 |
| 机器学习 | [2. 逻辑回归(Logistics Regression)](https://github.com/NLP-LOVE/ML-NLP/blob/master/Machine%20Learning/2.Logistics%20Regression/2.Logistics%20Regression.md) | [@mantchs](https://github.com/NLP-LOVE) | 448966528 |
| 机器学习 | [3. 决策树(Desision Tree)](https://github.com/NLP...

地址：[https://github.com/NLP-LOVE/ML-NLP](https://github.com/NLP-LOVE/ML-NLP)

---

## 🤩Python随身听-技术精选： /Pierian-Data/Complete-Python-3-Bootcamp
👉Course Files for Complete Python 3 Bootcamp Course on Udemy  

😎TOPICS: ``  

⭐️STARS：12636, 今日上升数↑:11  

👉README：

## Complete-Python-3-Bootcamp
Course Files for Complete Python 3 Bootcamp Course on Udemy

Get it now for ...

地址：[https://github.com/Pierian-Data/Complete-Python-3-Bootcamp](https://github.com/Pierian-Data/Complete-Python-3-Bootcamp)

---
